### PR TITLE
Add agent component overview document

### PR DESCRIPTION
**Pull request by vankichi**
Signed-off-by: vankichi <kyukawa315@gmail.com>

### Description:

I have created a document about the agent component overview.

### Environment:

- Go Version: 1.17.5
- Docker Version: 20.10.8
- Kubernetes Version: 1.22.0
- NGT Version: 1.13.8

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [x] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [ ] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [ ] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by kmrmt — APPROVED** [review]

**Review by kmrmt — CHANGES_REQUESTED** [review]
I'm sorry if `assets/docs/overview/component/agent/ngt.png` has already been discussed, but I think I need to add `Upsert` to the list of operations.
(I don't know how to comment to binary file directly...)

**Comment by vankichi** [comment]
@kmrmt 
> I'm sorry if `assets/docs/overview/component/agent/ngt.png` has already been discussed, but I think I need to add `Upsert` to the list of operations. (I don't know how to comment to binary file directly...)

This image applies to the features list, which I don't mention the `Upsert` in.
It is because a little bit complex comparing other functions.
If it should be there, I'll add it and update the image too.

**Review by kmrmt — APPROVED** [review]
@vankichi OK, I understand.

File: docs/overview/component/agent.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+The main functions are:
+
+- Backup
+  - When `Agent Core` completes creating the index metadata files, `Sidecar` hooks to store these to the external storage.
+
+<img src="../../../assets/docs/overview/component/agent/sidecar_backup.png" />
+
+- Restore
+  - When the Vald Agent Pod reboots, the index structure is restored using external backup files.
```
Comment (kevindiu) [inline docs/overview/component/agent.md:93]:
```suggestion
  - When the Vald Agent Pod restart, the index structure is restored from the external backup files.
```

File: docs/overview/component/agent.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  - Requested vectors are stored in the `vqueue`.
+- Delete
+  - Create a request to remove the specific vectors from NGT indexes.
+  - Requested vectors are stored in the `vqueue`.
+- GetObject
+  - Get the information of the indexed vectors.
+- Exist
+  - Check the specific vectors are already indexed or not.
+- CreateIndex
+  - Create a new NGT index structure in memory using vectors stored in the `vqueue` and the existing NGT index structure if it exists.
```
[the remark this answers is not in the document]
Comment (vankichi) [reply docs/overview/component/agent.md:57]: @kevindiu If NGT index structure already exists, the new NGT index consists of new insert vectors and it.

File: docs/overview/component/agent.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+You have to control the duration of CreateIndex and SaveIndex by configuration.
+
+These methods don’t always run when getting the request.
+
+</div>
+
+<div class="warning">
+As you see, Vald Agent NGT can only search the nearest neighbors from the NGT index.
+
+You have to wait to complete CreateIndex and SaveIndex functions.
```
[the remark this answers is not in the document]
Comment (vankichi) [reply docs/overview/component/agent.md:71]: I already fixed by Grammarly suggestion

File: docs/overview/component/agent.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+Mainly, it consists of 1 or 2 small components, Vald Core and Vald Agent sidecar, depending on the intended use.
+
+This page introduces the overview and features of Vald Agent.
+
+## Responsibility
+
+Vald Agent is responsible for:
+
+- Store index data along to the user requests
```
[the remark this answers is not in the document]
Comment (vankichi) [reply docs/overview/component/agent.md:13]: sorry, I'm not sure about your question.
could you please show detail?

[the remark this answers is not in the document]
Comment (kevindiu) [reply docs/overview/component/agent.md:13]: Sorry I mean there is no grammar of `along to the` in english, so i couldn't understand what this sentence means.

File: docs/overview/component/agent.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kevindiu) [reply docs/overview/component/agent.md:57]: I think it should be more clear.
```suggestion
  - Create a new NGT index structure in memory using vectors stored in the `vqueue` and store to the existing NGT index structure.
```

[the remark this answers is not in the document]
Comment (vankichi) [reply docs/overview/component/agent.md:57]: IMO, it is not correct because NGT index structure does not exist when the initial phase

**Review by kevindiu — APPROVED** [review]

**Review by kmrmt — APPROVED** [review]

**Review by kpango — APPROVED** [review]

**Pull request merged**